docs/src/main/asciidoc/sending-emails.adoc

```diff
@@ -0,0 +1,205 @@
+    mailer.send(Mail.withText("to@acme.org", "A simple email from quarkus", "This is my body"));
+    return Response.accepted().build();
+}
+
+@GET
+@Path("/async")
+public CompletionStage<Response> sendASimpleEmailAsync() {
+    return reactiveMailer.send(
+            Mail.withText("to@acme.org", "A reactive email from quarkus", "This is my body"))
+            .thenApply(x -> Response.accepted().build());
+}
+----
+
+You can create new `io.quarkus.mailer.Mail` instances from the constructor or from the `Mail.withText` and
+`Mail.withHtml` helper methods. The `Mail` instance lets you add recipients (to, cc, or bcc), set the subject,
+headers, sender (from) address...
+
+You can also send several `Mail` objects in one call:
+
+[source, java]
+----
+mailer.send(mail1, mail2, mail3);
+----
+
+== Sending attachments
+
+To send attachment, just use the `addAttachment` methods on the `io.quarkus.mailer.Mail` instance:
+
+[source,java]
+----
+@GET
+@Path("/attachment")
+public Response sendEmailWithAttachment() {
+    mailer.send(Mail.withText("to@acme.org", "An email from quarkus with attachment",
+            "This is my body")
+            .addAttachment("my-file.txt",
+                "content of my file".getBytes(), "text/plain"));
+    return Response.accepted().build();
+}
+----
+
+Attachments can be created from raw bytes (as shown in the snippet) or files.
+When using a file, the content is read asynchronously.
+
+== Sending HTML emails with inlined attachments
+
+When sending HTML email, you can add inlined attachments.
+For example, you can send an image with your email, and this email will be displayed in the mail content.
+
+[source, java]
+----
+@GET
+@Path("/html")
+public Response sendingHTML() {
+    String body = "<strong>Hello!</strong>" + "\n" +
+        "<p>Here is an image for you: <img src=\"cid:my-image@quarkus.io\"/></p>" +
+        "<p>Regards</p>";
+    mailer.send(Mail.withHtml("to@acme.org", "An email in HTML", body)
+        .addInlineAttachment("quarkus-logo.png",
+            new File("quarkus-logo.png"),
+            "image/png", "<my-image@quarkus.io>"));
+    return Response.accepted().build();
+}
+----
+
+Note the _content-id_ format and reference.
+When you create the inline attachment, the content-id must be structured as follows: `<id@domain>`.
+When referenced, for instance in the `src` attribute, use `cid:id@domain` (without the `<` and `>`).
+
+== Using the underlying Vert.x Mail Client
+
+The Quarkus Mailer is implemented on top of the https://vertx.io/docs/vertx-mail-client/java/[Vert.x Mail Client], providing an asynchronous and non-blocking way to send emails.
+You can inject the client directly:
+
+[source, java]
+----
+@Inject MailClient client;
+----
+
+Are exposed:
+
+* the Axle client (`io.vertx.axle.ext.mail.MailClient`), using `CompletionStage` and Reactive Streams `Publisher`
+* the RX Java 2 client (`io.vertx.reactivex.ext.mail.MailClient`)
+* the bare client (`io.vertx.ext.mail.MailClient`)
+
```

Comment:
Can you point to the Vert.x extension guide for further details about the different flavors and when they are useful?

Reply:
Done.